### ISSUE TITLE
Unable to create postgres on standby machine

### ISSUE THREAD

**Issue by Raves1209** [citusdata/pg_auto_failover#74]
getting below error when trying to setup secondary instance :

```
[postgres@lxlab1569252301 postgres_log]$ pg_autoctl create postgres --pgdata /postgres_data --monitor postgres://autoctl_node@172.29.142.26:50001/pg_auto_failover
14:10:38 INFO  Using --nodename "lxlab1569252301.devlab.lowes.com", which resolves to IP address "172.29.142.23"
14:10:38 INFO  Continuing from a previous `pg_autoctl create` failed attempt
14:10:38 INFO  PostgreSQL state at registration time was: PGDATA does not exists
14:10:38 INFO  FSM transition from "wait_standby" to "catchingup": The primary is now ready to accept a standby
14:10:38 INFO  The primary node returned by the monitor is lxlab1569252313.devlab.lowes.com:50001
14:10:38 INFO  Initialising PostgreSQL as a hot standby
14:10:38 INFO  Target directory exists: "/postgres_data", stopping PostgreSQL
14:10:38 INFO  pg_ctl: no server running

14:10:38 INFO  pg_ctl stop failed, but PostgreSQL is not running anyway
could not remove file or directory "//backup": Permission denied
14:10:38 ERROR Failed to remove directory "//backup": Permission denied
14:10:38 ERROR Failed initialise standby server, see above for details
14:10:38 ERROR Failed to transition from state "wait_standby" to state "catchingup", see above.
```

i dont know from its checking /backup folder .

successfully created monitor and primary instance.
Version is 11.2 and Os types is linux x86_64.
Followed below link for setup 
https://github.com/citusdata/pg_auto_failover

**Comment by adiep-cdpq** [comment]
@Raves1209 When initializing your slave node for the first time, your pgdata folder should be non-existing. In your case, the slave node shouldn't have /postgres_data when you execute your command.

**Comment by adiep-cdpq** [comment]
However, on the docs, it says the following : 

> 3. Initialize a secondary node from scratch
> 
> This happens when --pgdata (or the environment variable PGDATA) points to a non-existing or empty directory, and when the monitor registration call assigns the state WAIT_STANDBY in step 1.
> 
> In that case, the pg_autoctl create command steps through the initial states of registering a secondary server, which includes preparing the primary server PostgreSQL HBA rules and creating a replication slot.
> 
> When the command ends succesfully, a PostgreSQL secondary server has been created with pg_basebackup and is now started, catching-up to the primary server.

The slave node initializing doesn't work if the pgdata directory **is existing but empty**. I get the following error : 

> 19:25:09 INFO  Initialising PostgreSQL as a hot standby
> 19:25:09 INFO  Target directory exists: "/home/postgres/data-slave", stopping PostgreSQL
> 19:25:09 ERROR pg_ctl: directory "/home/postgres/data-slave" is not a database cluster directory
>

**Comment by DimCitus** [comment]
Hi,

I think the main problem here has to do with using a mount-point or top-level root directory for PGDATA. It's documented in PostgreSQL to avoid doing this. Please create a directory within a mount point and have `postgres` be the owner of it and then target a sub-directory of that for PGDATA.

See https://www.postgresql.org/docs/current/creating-cluster.html#CREATING-CLUSTER-MOUNT-POINTS that says:

> Many installations create their database clusters on file systems (volumes) other than the machine's “root” volume. If you choose to do this, it is not advisable to try to use the secondary volume's topmost directory (mount point) as the data directory. Best practice is to create a directory within the mount-point directory that is owned by the PostgreSQL user, and then create the data directory within that. This avoids permissions problems, particularly for operations such as pg_upgrade, and it also ensures clean failures if the secondary volume is taken offline.

**Comment by adiep-cdpq** [comment]
@DimCitus However, the same setup and commands work without any issue with the master node. It can be initialized even of the pgdata is existing. The problem occurs only on the slave node.

**Comment by changchichung** [comment]
I have the same problem today .
```
postgres@pg_secondary:~$ pg_autoctl create postgres --nodename pg_secondary --monitor postgres://autoctl_node@pg_monitor:5432/pg_auto_failover
02:50:04 INFO  Continuing from a previous `pg_autoctl create` failed attempt
02:50:04 INFO  PostgreSQL state at registration time was: PGDATA exists
02:50:04 INFO  FSM transition from "wait_standby" to "catchingup": The primary is now ready to accept a standby
02:50:04 INFO  The primary node returned by the monitor is pg_primary:5432
02:50:04 INFO  Initialising PostgreSQL as a hot standby
02:50:04 ERROR Failed to ensure empty directory "//backup": Permission denied
02:50:04 ERROR Failed initialise standby server, see above for details
02:50:04 ERROR Failed to transition from state "wait_standby" to state "catchingup", see above.
postgres@pg_secondary:~$
```

and  yes , my $PGDATA is set to /database .

I will try change PGDATA to somewhere else , instead of folder on rootfs.

**Comment by DimCitus** [comment]
I believe this bug has been fixed in 1.0.6, released today. Closing this issue. Please re-open if the bug is still there, with enough information for us to reproduce and fix. Thanks.
